tests/backup/backup_test.go

```diff
@@ -237,6 +238,158 @@ var _ = Describe("{CreateMultipleUsersAndGroups}", func() {
+
+		})
+		Step("Adding Credentials and Registering Backup Location", func() {
+			log.InfoD("Creating cloud credentials and backup location")
+			for _, provider := range providers {
+				BucketName = fmt.Sprintf("%s-%v", provider, buckets[0])
+				cloudCredUID = uuid.New()
+				CloudCredUidList = append(CloudCredUidList, cloudCredUID)
+				backupLocationUID = uuid.New()
+				credName = fmt.Sprintf("autogenerated-cred-%v", time.Now().Unix())
+				CreateCloudCredential(provider, credName, cloudCredUID, orgID)
+				log.InfoD("Created Cloud Credentials with name - %s", credName)
+				time.Sleep(time.Minute * 1)
+				BackupLocationName = fmt.Sprintf("autogenerated-backup-location-%v", time.Now().Unix())
+				CreateBackupLocation(provider, BackupLocationName, backupLocationUID, credName, cloudCredUID, BucketName, orgID, "")
+				log.InfoD("Created Backup Location with name - %s", BackupLocationName)
+			}
+		})
+		Step("Register source and destination cluster for backup", func() {
+			log.InfoD("Registering Source and Destination clusters and verifying the status")
+			CreateSourceAndDestClusters(orgID, "", "", ctx)
+			clusterStatus, clusterUid = Inst().Backup.RegisterBackupCluster(orgID, SourceClusterName, "")
+			dash.VerifyFatal(clusterStatus, api.ClusterInfo_StatusInfo_Online, "Verifying backup cluster status")
+		})
+
+		Step("Taking backup of applications", func() {
+			err = CreateBackup(backupName, SourceClusterName, BackupLocationName, backupLocationUID, []string{bkpNamespaces[0]},
+				nil, orgID, clusterUid, "", "", "", "", ctx)
+			dash.VerifyFatal(err, nil, "Verifying backup creation")
+		})
+
+		Step("Share backup with user", func() {
+			log.InfoD("Share backup with  user having full access")
+			err := ShareBackup(backupName, nil, []string{userName}, FullAccess, ctx)
+			log.FailOnError(err, "Failed to share backup %s", backupName)
+		})
+
+		Step("Duplicate shared backup", func() {
+
+			// Get user context
+			ctxNonAdmin, err := backup.GetNonAdminCtx(userName, password)
+			log.FailOnError(err, "Fetching non px-central-admin user ctx")
+			userContexts = append(userContexts, ctxNonAdmin)
+
+			// Validate that backups are shared with user
+			log.Infof("Validating that backups are shared with %s user", userName)
+			userBackups1, _ := GetAllBackupsForUser(userName, password)
+			dash.VerifyFatal(len(userBackups1), numberOfBackups, fmt.Sprintf("Validating that user [%s] has access to all shared backups", userName))
+
+			//to duplicate shared backup internally it calls create backup api
+			log.Infof("Duplicate shared backup")
+			err = CreateBackup(backupName, SourceClusterName, BackupLocationName, backupLocationUID, []string{bkpNamespaces[0]},
+				nil, orgID, clusterUid, "", "", "", "", ctxNonAdmin)
+			log.Infof("user not able to duplicate shared backup without adding cluster with err - %v", err)
+			//Test will pass if error is not nil
+			gomega.Expect(err).ToNot(gomega.BeNil(), "Verifying Duplicate shared ")
```

Comment:
No need to use gomega library. Just use `err` and validate it against the error text.
For example in case of user trying to delete a backup for which the user has View Only access, we are validating the error like this

```
// Delete backup to confirm that the user cannot delete the backup
_, err = DeleteBackup(backupName, backupUID, orgID, ctxNonAdmin)
log.Infof("Error message - %s", err.Error())
dash.VerifyFatal(strings.Contains(err.Error(), "doesn't have permission to delete backup"), true, "Verifying backup deletion is not possible")
```

This is just an example. Please see what text you get in your error message from the API call

Reply:
Thanks for the reference , I've replaces the gomega library